forwarder/router_client.go

```diff
@@ -380,10 +385,35 @@ func (rc *RouterClient) run(ctx context.Context) error {
 					return nil
 				}
 			}
+		case <-time.After(joinFilterRenewInterval):
```

Comment:
You want to use a `time.Ticker` here. `time.After` resets on each iteration of the loop.